finesse/hardware/temperature/dp9800.py

```diff
@@ -0,0 +1,238 @@
+
+    def get_device_settings(self, sysflag: str) -> dict[str, str]:
+        """Provide the settings of the device as stored in the system flag.
+
+        The system flag is stored as a bit mask with the format TxxLxSAF,
+        where:
+            F - bit 0: temperature unit: 0 = deg C, 1 = deg F
+            A - bit 1: audible button:   0 = silence, 1 = audible
+            S - bit 2: autoscan:         0 = no scan, 1 = autoscan active
+            x - bit 3: must be 0
+            L - bit 4: logging:          0 = no logging, 1 = logging active
+            x - bit 5: must be 0
+            x - bit 6: must be 0
+            T - bit 7: instrument type:  0 = TC, 1 = PT
+
+        Args:
+            sysflag: string representation of the system flag bitmask
+
+        Returns:
+            dev_settings: dictionary containing the current device settings
+        """
+        dev_settings = {
+            "instrument_type": ["TC", "PT"][int(sysflag[0])],
+            "logging_state": ["no logging", "logging active"][int(sysflag[3])],
+            "scanning_state": ["no scan", "autoscan active"][int(sysflag[5])],
+            "audible_state": ["silence", "audible"][int(sysflag[6])],
+            "temperature_unit": ["deg C", "deg F"][int(sysflag[7])],
+        }
+        return dev_settings
+
+    def read(self) -> bytes:
+        """Read temperature data from the DP9800.
+
+        The DP9800 returns a sequence of bytes containing the
+        temperatures measured on each channel, in the format
+
+          STX T SP t1 SP t2 SP t3 SP t4 SP t5 SP t6 SP t7 SP t8 SP t9 ff ETX BCC NUL
+
+        where
+            t1, t2, ..., t9: temperature values in the format %4.2f
+            STX: Start of Text (ASCII 2)
+            ETX: End of Text (ASCII 3)
+            NUL: Null character (ASCII 0)
+            SP: Space (ASCII 32)
+            BCC: Block Check Character
+            ff: System flag in hexadecimal
+
+        For error checking, the BCC is calculated by performing consecutive XOR
+        operations on the message and compared with the BCC received.
+
+        Note a slight peculiarity:
+        The device actually returns 9 temperatures, while the documentation
+        states that there should be 8. The device shows 8 values, corresponding
+        to the values with indices 1 to 8. The first value (at index 0) is
+        therefore ignored.
+
+        Returns:
+            data: the sequence of bytes read from the device
+
+        Raises:
+            DP9800Error: Malformed message received from device
+        """
+        # require at least 4 bytes else checks will fail
+        min_length = 4
+        try:
+            data = self.serial.read_until(b"\x00", size=min_length)
+        except SerialException as e:
+            raise DP9800Error(e)
+
+        if len(data) < min_length:
+            raise DP9800Error("Insufficient data read from device")
+        else:
+            return data
```

Comment:
You don't need `else` here:

```suggestion
        return data
```